src/containerbuddy/ips.go

```diff
@@ -0,0 +1,253 @@
+	IPv6    bool
+	Name    string
+	Network *net.IPNet
+	Index   int
+}
+
+func (spec interfaceSpec) String() string {
+	return spec.Spec
+}
+
+func (spec interfaceSpec) Match(index int, iip interfaceIP) bool {
+	// Specific Interface eth1, eth0[1], eth0:inet6, inet, inet6
+	if spec.Name == iip.Name || spec.Name == "*" {
+		// Has index and matches
+		if spec.Index > 0 {
+			return (spec.Index == index)
+		}
+		// Don't match loopback address for wildcard spec
+		if spec.Name == "*" && iip.IP.IsLoopback() {
+			return false
+		}
+		return spec.IPv6 != iip.IsIPv4()
+	}
+	// CIDR
+	if spec.Network != nil && spec.Network.Contains(iip.IP) {
+		return true
+	}
+
+	return false
+}
+
+func parseInterfaceSpecs(interfaces []string) ([]interfaceSpec, error) {
+	var errors []string
+	var specs []interfaceSpec
+	for _, iface := range interfaces {
+		spec, err := parseInterfaceSpec(iface)
+		if err != nil {
+			errors = append(errors, err.Error())
+			continue
+		}
+		specs = append(specs, spec)
+	}
+	if len(errors) > 0 {
+		err := fmt.Errorf(strings.Join(errors, "\n"))
+		println(err.Error())
+		return specs, err
+	}
+	return specs, nil
+}
+
+var (
+	ifaceSpec = regexp.MustCompile(`^(?P<Name>\w+)(?:(?:\[(?P<Index>\d+)\])|(?::(?P<Version>inet6?)))?$`)
+)
+
+func parseInterfaceSpec(spec string) (interfaceSpec, error) {
+	if spec == "inet" {
+		return interfaceSpec{IPv6: false, Name: "*"}, nil
+	}
+	if spec == "inet6" {
+		return interfaceSpec{IPv6: true, Name: "*"}, nil
+	}
+
+	if match := ifaceSpec.FindStringSubmatch(spec); match != nil {
+		name := match[1]
+		index := match[2]
+		inet := match[3]
+		if index != "" {
+			i, _ := strconv.Atoi(index)
+			if i > 0 {
+				return interfaceSpec{Spec: spec, Name: name, Index: i}, nil
+			}
+			return interfaceSpec{Spec: spec}, fmt.Errorf("Unable to parse interface spec: %s. Index should be > 0", spec)
+		}
+		if inet != "" {
+			if inet == "inet" {
+				return interfaceSpec{Spec: spec, Name: name, IPv6: false}, nil
+			}
+			return interfaceSpec{Spec: spec, Name: name, IPv6: true}, nil
+		}
+		return interfaceSpec{Spec: spec, Name: name, IPv6: false}, nil
+	}
+	if _, net, err := net.ParseCIDR(spec); err == nil {
+		return interfaceSpec{Spec: spec, Network: net}, nil
+	}
+	return interfaceSpec{Spec: spec}, fmt.Errorf("Unable to parse interface spec: %s", spec)
+}
+
+type interfaceIP struct {
+	Name string
+	IP   net.IP
+}
+
+func (iip interfaceIP) To16() net.IP {
+	return iip.IP.To16()
+}
+
+func (iip interfaceIP) To4() net.IP {
+	return iip.IP.To4()
+}
+
+func (iip interfaceIP) IsIPv4() bool {
+	return iip.To4() != nil
+}
+
+func (iip interfaceIP) IPString() string {
+	if v4 := iip.To4(); v4 != nil {
+		return v4.String()
+	}
+	return iip.IP.String()
+}
+
+func (iip interfaceIP) String() string {
+	return fmt.Sprintf("%s:%s", iip.Name, iip.IPString())
+}
+
+// Queries the network interfaces on the running machine and returns a list
+// of IPs for each interface. Currently, this only returns IPv4 addresses.
```

Comment:
I believe this comment is now stale: `Currently, this only returns IPv4 addresses.` ?